File: DataImportManager/clsDBTask.vb
Imports System.Collections.Specialized
Imports System.Data.SqlClient
Imports PRISM.Logging

Public MustInherit Class clsDBTask

#Region "Member Variables"

    ' access to the logger
    Protected ReadOnly m_logger As ILogger

    ' access to mgr parameters
    Protected ReadOnly m_mgrParams As IMgrParams

    ' DB access
    Protected ReadOnly m_DBCn As SqlConnection

#End Region

#Region "Auto-properties"
    Public Property TraceMode As Boolean
#End Region

    ''' <summary>
    ''' Constructor
    ''' </summary>
    ''' <param name="mgrParams"></param>
    ''' <param name="logger"></param>
    ''' <param name="dbConnection">Database connection object (connection should already be open)</param>
    ''' <remarks></remarks>
    Public Sub New(mgrParams As IMgrParams, logger As ILogger, dbConnection As SqlConnection)
        m_mgrParams = mgrParams
        m_logger = logger
        m_DBCn = dbConnection
    End Sub

End Class

File: DataImportManager/clsDBTask.cs
using System.Data.SqlClient;
using PRISM.AppSettings;

namespace DataImportManager
{
    // ReSharper disable once InconsistentNaming
    internal abstract class clsDBTask : clsLoggerBase
    {
        /// <summary>
        /// Manager parameters
        /// </summary>
        protected MgrSettings MgrParams { get; }

        /// <summary>
        /// Database connection object
        /// </summary>
        protected SqlConnection DatabaseConnection { get; }

        /// <summary>
        /// When true, show additional debug messages as the console
        /// </summary>
        public bool TraceMode { get; set; }

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="mgrParams">Manager parameters</param>
        /// <param name="dbConnection">Database connection object (connection should already be open)</param>
        protected clsDBTask(MgrSettings mgrParams, SqlConnection dbConnection)
        {
            MgrParams = mgrParams;
            DatabaseConnection = dbConnection;
        }
    }
}
